core/shared/src/test/scala/fs2/concurrent/SignalSuite.scala

```diff
@@ -320,6 +321,19 @@ class SignalSuite extends Fs2Suite {
     TestControl.executeEmbed(prog).assertEquals(expected)
   }
 
+  test("SignallingRef.mapK() returns a SignallingRef") {
+    for {
+      s <- SignallingRef[IO, Int](0)
+      nt = new FunctionK[IO, IO] {
+        def apply[A](fa: IO[A]): IO[A] = fa
+      }
+      transformed = s.mapK(nt)
```

Comment:
Let's also check that the type infers correctly at compile-time.

```suggestion
      transformed: SignallingRef[IO, Int] = s.mapK(nt)
```